query-engine/connectors/query-connector/src/interface.rs

```diff
@@ -12,6 +12,15 @@ pub trait Connector {
     /// Returns the name of the connector.
     fn name(&self) -> &'static str;
 
+    /// The database system name, as per the OTEL spec.
+    /// In most cases, this is the same as the connector's name.
+    /// Reference:
+    /// - https://opentelemetry.io/docs/specs/semconv/database/sql/
+    /// - https://opentelemetry.io/docs/specs/semconv/database/mongodb/
+    fn system_name(&self) -> &'static str {
+        self.name()
+    }
```

Comment:
This is only used in `query-engine/core/src/executor/execute_operation.rs`.
For other metrics spans, the logic lives in:
- `provider = "mongodb": `query-engine/connectors/mongodb-query-connector/src/root_queries/mod.rs`
- SQL providers: their corresponding `quaint` native module.